### PR TITLE
Faster Zero123 convergence

### PR DESCRIPTION
**Pull request by claforte**
My goal was to get a good-enough reconstruction of Anya in 500-1000 iterations using Zero123 and to avoid Janus artifacts. It doesn't need to be perfect - later training phases will use guidance from SD, DeepFloyd etc. to correct the imperfections.

To reproduce, run the last script in `threestudio/scripts/run_zero123_examples.sh`.

Results after 1999 steps (default for my last script, takes ~35mins on 1 A100 40GB):

https://github.com/threestudio-project/threestudio/assets/5090329/cd10e647-d203-4e49-be5a-9d10305b10ea

... after ~1000 steps:

https://github.com/threestudio-project/threestudio/assets/5090329/873464af-4a20-4c5b-9a02-b48da1bb9f8d

... after ~500 steps:

https://github.com/threestudio-project/threestudio/assets/5090329/92f14d44-1c10-43e9-b123-c4a60dc32538

training loss over time:

image

NOTE that I ran my experiments with a yet-to-be-released Zero123XL provided by @ruoshiliu. I think it should still work with the normal Zero123 as well, but may take longer to converge.

Main changes:
- support wandb run name, editable through command line
- much higher learning rate - converges much faster but relies on higher batch size
- increase batch size to 4, to fit within 40GB. (you can try decreasing the batch size if you don't have sufficient VRAM, but you'll probably need to decrease the lr as well to compensate)
- accumulate mode (instead of alternate image conditioning and Zero123-guidance) seems to converge slightly faster and produce slightly less bumpy loss, on Anya, comparing wall clock. (however each step runs ~80% slower)
- randomize background color per-item. (Previously, ~50% of batches were white, 50% of batches had 1 same random color applied to every batch item.) This makes a small difference for some of the loss terms, and a tiny improvement in convergence speed.
   - NOTE: this has a **side effect**. **Validation and test videos used to always have a white background. Now the background has random colors, as shown above.**
- slightly narrowed the noise pct range from [0.02, 0.98] to [0.1, 0.9]. This made minor but noticeable improvements in final (1999 steps) 3D reconstruction (e.g. fingers are better defined). Previously, very high noise level (~ >0.9) resulted in poor Zero123 predictions so I estimate that 10% of the training effort was wasted. Unfortunately huge spikes in the loss (when adding a new hash grid level?) caused huge spikes in some loss terms (see above). In comparison, this was the loss, with the full noise range and the rest of the changes mentioned here:

image

### CODE REVIEW

**Comment by claforte** [comment]
I recommend squash-and-merge. @bennyguo I think this should be safe and isolated to Zero123, but you might want to double-check the small changes in `threestudio/models/renderers/nerf_volume_renderer.py` to support different background colors for each batch item.

File: threestudio/models/renderers/nerf_volume_renderer.py [hunk 1]
```diff
@@ -162,6 +162,13 @@ def forward(
...
             bg_color = comp_rgb_bg
         else:
+            if bg_color.shape == (batch_size, 3):
+                # e.g. constant random color used for Zero123
+                # [bs,3] -> [bs, 1, 1, 3]):
+                bg_color = bg_color.unsqueeze(1).unsqueeze(1)
+                #        -> [bs, height, width, 3]):
+                bg_color = bg_color.expand(-1, height, width, -1)
+
             if bg_color.shape == (batch_size, height, width, 3):
```
Comment (voletiv) [inline threestudio/models/renderers/nerf_volume_renderer.py:172]: Doesn't this undo the above change?

File: configs/zero123.yaml [hunk 2]
```diff
@@ -116,6 +116,7 @@ system:
     wandb:
       enable: false
       project: "threestudio"
+      name: None
```
Comment (voletiv) [inline configs/zero123.yaml:119]: Also, ref_or_zero123 needs to be `accumulate`?

**Comment by voletiv** [comment]
Looks good to me, most changes are in bg_color, and wandb name. Please see my comments :)

File: configs/zero123.yaml (hunk 2 above)
Comment (voletiv) [reply configs/zero123.yaml:119]: Also, higher LR?

Comment (claforte) [reply configs/zero123.yaml:119]: I didn't want to override the defaults just yet... for now you need to run my shell script (which does a bunch of overrides through the command line)

File: threestudio/models/renderers/nerf_volume_renderer.py (hunk 1 above)
Comment (voletiv) [reply threestudio/models/renderers/nerf_volume_renderer.py:172]: Figured it out, thanks!

**Review by voletiv — APPROVED** [review]

File: configs/zero123.yaml (hunk 2 above)
Comment (claforte) [reply configs/zero123.yaml:119]: Later, when we test on more objects, if we see that this set of config changes consistently outperforms the current default, we can confidently change the default. ;-)
